### DISCUSSION TITLE
How to get current frame data & index?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1191] [Q&A]
Hi:

Is it possible to get the current frame data like which frame(1st, 2nd or 3rd) is currently showing
in the onUpdate method?
We tried to use method like viewComponent.getAnimationChannel().getCurrentFrame() but failed

Thanks

**Comment by lhDream** [comment]
@chengenzhao 每次onUpdate被执行的时候都是新的一帧,也许我们可以自己算...

**Reply by chengenzhao** [reply]
我现在就是加上一个counter，然后每次update+1，自己算，但是感觉不够优雅

**Comment by AlmasB** [comment]
Hi, do you mean this in AnimatedTexture?

```
val frameData = animationChannel.getFrameData(currentFrame)
```

It would be quite easy to make this public. Can you provide your use case please.

**Reply by chengenzhao** [reply]
yeah
sth like this
for example we have a floating unit
which needs to adjust its vertical position based on the frame
e.g. the 1st flame translate y + 2, 2nd +1, 3rd 0, 4th -1, 5th -2, 6th -1, 7th 0, 8th +1, 9th +2, 10th +1, 11th 0, 12th -1 ...

**Reply by chengenzhao** [reply]
And FrameData seems doesn't provide which frame(e.g. the 3rd frame) the current frame is?

**Reply by chengenzhao** [reply]
Hi @AlmasB :

I just checked the source code, and seems like
It should be easy to make getCurrentFrame data public?
just add get & set for it, then the mechanism will handle the rest
would you mind opening this variable?
This time we want to make a instant copy of entity
and for view component especially for animated texture
this method could help developers decide which frame is using(get) and also use set to set the start the animation from a specific frame, I think this is another requirement we have discussed before?
so this would help solve two problems at the same time.
![截屏2023-03-04 14 54 25](https://user-images.githubusercontent.com/5525436/222880924-966f5d7b-b052-4824-bd87-2d7a3e88d48c.png)
